modules/json/src/lib/json-configuration.js

```diff
@@ -0,0 +1,35 @@
+// TODO - default parsing code should not be part of the configuration.
+import parseExpressionString from './helpers/parse-expression-string';
+import assert from '../utils/assert';
+const DEFAULT_TYPE_KEY = 'type';
+
+const DEFAULT_CONFIGURATION = {
+  typeKey: DEFAULT_TYPE_KEY,
+  classes: {},
+  reactComponents: {},
+  enumerations: {},
+  // TODO - this needs to be simpler, function conversion should be built in
+  convertFunction
+};
+
+const isObject = value => value && typeof value === 'object';
+
+export default class JSONConfiguration {
+  constructor(configuration) {
+    Object.assign(this, DEFAULT_CONFIGURATION, configuration);
+  }
+
+  validate(configuration) {
+    assert(!this.typeKey || typeof this.typeKey === 'string');
```

Comment:
Will these produce any messages explaining what didn't validate and why the failures happened?

Reply:
> Will these assert statements produce any messages explaining what didn't validate / why the failures happened?

Good question: No - I usually avoid adding messages for "programmer errors", since they add to the bundle size but provide very little value as a programmer can just break on the exception and see what went wrong.

For data validation errors that affect the user, I agree we should provide generous error message.